### PR TITLE
update kustomize template to fit for latest kustomize

### LINKED ISSUE #356: support latest kustomize
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
we only support kustomize 1.0.11, at least we need know the reason and see what we can do to support latest one...

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by jichenjc**
commands is removed and replaced with files and literals
so update template accordingly

https://github.com/kubernetes-sigs/kustomize/commit/2fa4a3458924de6c3a1613fb0c69b2f658e6b3c7 has more info

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #356 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hidekazuna** [comment]
Could you update update kustomize version in the followings?

- Cluster Creation section in README.md
- cmd/clusterctl/examples/openstack/README.md

**Comment by jichenjc** [comment]
yes, we should, thanks for the reminder

File: cmd/clusterctl/examples/openstack/README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 1. Install `yq` (see [here](https://github.com/mikefarah/yq)).
 2. Install `kustomize` v1.0.11 which can be found [here](https://github.com/kubernetes-sigs/kustomize/releases/tag/v1.0.11).
```
Comment (sbueringer) [inline cmd/clusterctl/examples/openstack/README.md]: Change to kustomize v2.0.3

File: cmd/clusterctl/examples/openstack/provider-component/cluster-api/kustomization.yaml [hunk 2]
```diff
@@ -4,14 +4,14 @@ generatorOptions:
   disableNameSuffixHash: true
```
Comment (sbueringer) [inline cmd/clusterctl/examples/openstack/provider-component/cluster-api/kustomization.yaml]: I replaced the resources block through the following. This includes the kustomization files in this folder which should lead to the same result (but we don't need the cp in generate-yaml.sh):
````
bases:
-  ../../../../../../vendor/sigs.k8s.io/cluster-api/config/rbac
- ../../../../../../vendor/sigs.k8s.io/cluster-api/config/crds
- ../../../../../../vendor/sigs.k8s.io/cluster-api/config/manager
````

File: cmd/clusterctl/examples/openstack/README.md (hunk 1 above)
Comment (jichenjc) [reply cmd/clusterctl/examples/openstack/README.md]: ok, will do

File: cmd/clusterctl/examples/openstack/provider-component/cluster-api/kustomization.yaml (hunk 2 above)
Comment (jichenjc) [reply cmd/clusterctl/examples/openstack/provider-component/cluster-api/kustomization.yaml]: the issue is I tried this and I was informed by kustomize that include files outside the folder violate the security restriction, so I have to copy the files...

Comment (sbueringer) [reply cmd/clusterctl/examples/openstack/provider-component/cluster-api/kustomization.yaml]: I have to check. That was in the release notes
> In 2.0, bases can continue to specify, via relative paths, kustomizations outside the current kustomization directory. But non-base paths are constrained to terminate in or below the current kustomization directory. 

https://github.com/kubernetes-sigs/kustomize/blob/master/docs/v_2.0.0.md

And it looked like it worked

File: cmd/clusterctl/examples/openstack/generate-yaml.sh [hunk 3]
```diff
@@ -234,6 +234,12 @@ fi
 echo "---" >> $PWD/$OUTPUT/provider-components.yaml
 kustomize build $PWD/provider-component/clouds-secrets >> $PWD/$OUTPUT/provider-components.yaml
 echo "---" >> $PWD/$OUTPUT/provider-components.yaml
+
+# latest kustomize don't allow include files out of build folder
+cp -r $CONFIG_DIR/../../../../../../../vendor/sigs.k8s.io/cluster-api/config $PWD/provider-component/cluster-api
+echo "---" >> $PWD/$OUTPUT/provider-components.yaml
 kustomize build $PWD/provider-component/cluster-api >> $PWD/$OUTPUT/provider-components.yaml
```
Comment (sbueringer) [inline cmd/clusterctl/examples/openstack/generate-yaml.sh]: Should we change these calls to `kubectl kustomize` so we Font need the kustomize binary?

Comment (jichenjc) [reply cmd/clusterctl/examples/openstack/generate-yaml.sh]: yes, I will update

**Comment by pyama86** [comment]
This is a great pull request. I am looking forward to Merge!

**Comment by jichenjc** [comment]
@chrigl  @xunpan @gyliu513 help to take a look?

File: Makefile [hunk 4]
```diff
@@ -82,9 +82,8 @@ ifndef HAS_YQ
 endif
 
 ifndef HAS_KUSTOMIZE
```
Comment (sbueringer) [inline Makefile]: Do we still need this in the Makefile if we're using kubectl?

Comment (jichenjc) [reply Makefile]: yes, I removed them, thanks for the reminder

File: README.md [hunk 5]
```diff
@@ -72,7 +72,7 @@ policy may be made to more closely align with other providers in the Cluster API
 
 ### Prerequisites
 
-1. Install `kubectl` (see [here](http://kubernetes.io/docs/user-guide/prereqs/)).
+1. Install `kubectl` (see [here](http://kubernetes.io/docs/user-guide/prereqs/)). Because `kustomize` was included into `kubectl` and it's used by `cluster-api-provider-openstack` in generating yaml files, so version `1.14.0+` of `kubectl` is required, see [integrate kustomize into kubectl](https://github.com/kubernetes/enhancements/issues/633) for more info.
```
Comment (gyliu513) [inline README.md]: Can you update the link to https://kubernetes.io/docs/tasks/tools/install-kubectl/

**Pull request merged**